src/hosted-form/hosted-form-order-data-transformer.ts

```diff
@@ -0,0 +1,41 @@
+import { omit } from 'lodash';
+
+import { ReadableCheckoutStore } from '../checkout';
+import { MissingDataError, MissingDataErrorType } from '../common/error/errors';
+import { OrderPaymentRequestBody } from '../order';
+import { HostedCreditCardInstrument } from '../payment';
+
+import HostedFormOrderData from './hosted-form-order-data';
+
+export default class HostedFormOrderDataTransformer {
+    constructor(
+        private _store: ReadableCheckoutStore
+    ) {}
+
+    transform(payload: OrderPaymentRequestBody): HostedFormOrderData {
```

Comment:
I wonder if this is just a transformer, since it also takes the store as an argument, it's more like `HostedFormOrderDataBuilder`? 🍹

Reply:
I had a chat with Luis regarding this and agreed that naming it as `builder` might be confusing as the implementation doesn't really follow the builder pattern. In effect, the class is responsible for transforming one type of payload data object into another. So it seems `transformer` might the most appropriate. I can revisit this if I can find a better name for this in the future. For now, we're happy to proceed with what I have.